src/App.js

```diff
@@ -1,5 +1,133 @@
+
 class App {
-  async play() {}
+  async play() {
+    Console.print('숫자 야구 게임을 시작합니다.');
+
+    while (true) {
+      await this.gameFlow();
+      const userRestartInput = await this.getUserRestartInput();
+      if (userRestartInput !== 1) {
+        break;
+      }
+    }
+  }
+  // 게임 실행 흐름을 책임지는 함수
+  async gameFlow() {
+    const COMPUTER_ANSWER = this.createComputerAnswer();
+    let isUserWon = false;
+
+    while (!isUserWon) {
+      isUserWon = await this.checkUserGuess(COMPUTER_ANSWER);
+    }
+    return;
+  }
+
+  // 사용자 입력을 받는 getUserInput
+  async getUserInput() {
+    try {
+      const answer = await Console.readLineAsync('숫자를 입력해주세요 : ');
+
+      const userInput = answer.split('');
+
+      if (
+        userInput.length !== 3 ||
+        userInput.some(char => isNaN(parseInt(char, 10))) ||
+        new Set(userInput).size !== 3
+      ) {
+        throw new Error(
+          '[ERROR] 1에서 9 사이의 서로 다른 세 숫자를 입력해 주세요.'
+        );
+      }
+
+      return userInput.map(char => parseInt(char, 10));
+    } catch (error) {
+      throw new Error(error.message);
+    }
+  }
+
+  // 랜덤한 컴퓨터 정답을 생성하는 함수
+  createComputerAnswer = () => {
+    let computerAnswer = [];
+    while (computerAnswer.length < 3) {
+      let randomNum = Random.pickNumberInRange(1, 9);
+
+      if (!computerAnswer.includes(randomNum)) {
+        computerAnswer.push(randomNum);
+      }
+    }
+    return computerAnswer;
+  };
+
+  // 낫싱을 확인하는 함수
+  isNothing = (userGuess, computerAnswer) => {
+    return userGuess.filter(x => computerAnswer.includes(x)).length === 0
+      ? true
+      : false;
+  };
+
+  // 스트라이크와 볼을 확인하는 함수
+  checkStrikeOrBall = (userGuess, computerAnswer) => {
+    let strikeCount = 0,
+      ballCount = 0;
+
+    for (let i = 0; i < userGuess.length; i++) {
+      if (userGuess[i] === computerAnswer[i]) {
+        strikeCount += 1;
+      } else if (computerAnswer.includes(userGuess[i])) {
+        ballCount += 1;
+      }
+    }
+
+    return { strikeCount, ballCount };
+  };
+
+  // 사용자의 추측을 출력하는 함수
+  printGuessResult = guessResult => {
+    if (guessResult.strikeCount === 3) {
+      Console.print('3스트라이크\n3개의 숫자를 모두 맞히셨습니다! 게임 종료');
+    } else if (guessResult.strikeCount === 0) {
+      Console.print(`${guessResult.ballCount}볼`);
+    } else if (guessResult.ballCount === 0) {
+      Console.print(`${guessResult.strikeCount}스트라이크`);
+    } else
+      Console.print(
+        `${guessResult.ballCount}볼 ${guessResult.strikeCount}스트라이크`
+      );
+  };
+
+  // 사용자의 입력을 받고 체크하는 과정의 함수
+  async checkUserGuess(computerAnswer) {
+    let userGuess = await this.getUserInput();
+    if (this.isNothing(userGuess, computerAnswer)) {
+      Console.print('낫싱');
+      return false;
+    } else {
+      let guessResult = this.checkStrikeOrBall(userGuess, computerAnswer);
+      this.printGuessResult(guessResult);
+      if (guessResult.strikeCount === 3) {
+        return true;
+      }
+    }
+    return false;
+  }
```

Comment:
낫싱도 결과 문구의 일종이라고 생각해서 `getResultString` 함수 안에 같이 작성하면 어떨까요?
`checkUserGuess` 함수가 유저가 정답을 맞췄는지 결과를 반환해야 하기 때문에
일부러 `isNothing` 함수를 만들어서 맞추지 못했을 경우에 false 를 반환하는 로직을 추가해주신 것 같아요..!
그런데 strike===3 이 아닌 경우( 1볼 1스트라이크, 낫싱, 2볼 등등 ), 즉 정답을 맞추지 못했을 경우는
모두 마지막 return false 에서 함께 반환될 때니 굳이 따로 빼서 return 해주지 않아도 될 것 같은데 따로 빼주신 이유가 있을까요??

```suggestion
  // 사용자의 입력을 받고 체크하는 과정의 함수
  async checkUserGuess(computerAnswer) {
    let userGuess = await this.getUserInput();
     let guessResult = this.checkStrikeOrBall(userGuess, computerAnswer);
      this.printGuessResult(guessResult); // 내부적으로 낫싱 출력하는 로직 추가
      if (guessResult.strikeCount === 3) {
        return true;
      }
    }
    return false; // 정답을 맞추지 못한 모든 경우
  }
```